src/main/java/io/anserini/rerank/lib/NewsBackgroundLinkingReranker.java

```diff
@@ -54,29 +55,43 @@ public ScoredDocuments rerank(ScoredDocuments docs, RerankerContext context) {
     }
     
     // remove the duplicates: 1. the same doc with the query doc 2. duplicated docs in the results
-    Set<Integer> duplicates = new HashSet<>();
+    Set<Integer> toRemove = new HashSet<>();
     for (int i = 0; i < docs.documents.length; i++) {
-      if (duplicates.contains(i)) continue;
-      String docid = docs.documents[i].getField(FIELD_ID).stringValue();
+      if (toRemove.contains(i)) continue;
       if (computeCosineSimilarity(queryTermsMap, docsVectorsMap.get(i)) >= 0.9) {
-        duplicates.add(i);
+        toRemove.add(i);
         continue;
       }
       for (int j = i+1; j < docs.documents.length; j++) {
         if (computeCosineSimilarity(docsVectorsMap.get(i), docsVectorsMap.get(j)) >= 0.9) {
-          duplicates.add(j);
+          toRemove.add(j);
+        }
+      }
+    }
+
+    if(context.getSearchArgs().backgroundlinking_datefilter){
+      try{
+        Document queryDoc = reader.document(NewsBackgroundLinkingTopicReader.convertDocidToLuceneDocid(reader, queryDocId));
```

Comment:
You can use `LongPoint.newRangeQuery` here. 
See http://lucene.apache.org/core/8_2_0/core/org/apache/lucene/document/LongPoint.html#newRangeQuery-java.lang.String-long-long-

Reply:
I will try this, but I expect if you do it in the initial retrieval step the results will be different as the RM3 constructed query will be different.

Reply:
This is actually interesting. 
I think we should keep the filter same for both initial ranking and the relevance feedback otherwise the results are biased. right?

Reply:
If the filter models "no information from the future should be used", then the filter should be everywhere (and technically, IDF values should also depend on time...).

If it is just reflecting how judging took place, then maybe that is a different situation.

Reply:
How about let's make this consistent with what we did for microblog track? 
That is, to only filter the "future" documents in the initial ranking? 
https://github.com/castorini/anserini/blob/453a9d0307bfb331e83a8d9c48eef269379ccefb/src/main/java/io/anserini/search/SearchCollection.java#L505

Reply:
Ok the results are:

BM25 with datefilter in initial ranking:
`NDCG@5 : 0.3735`

BM25 + RM3 with datefilter in initial ranking:
`NDCG@5 : 0.3433`

BM25 + RM3 with datefilter in both initial ranking and reranking:
`NDCG@5 : 0.3444`

BM25 + RM3 with datefilter in only reranking (initial implementation)
`NDCG@5 : 0.4171`

Reply:
So it seems that datefilter in initial ranking is good when no reranking is applied, however when trying to rerank the query construction of RM3 does suffer from missing the documents that are filtered out.

Reply:
Well, it depends on the answer to my question below about the task guidelines... do we want to "win" or model the task accurately? :)